### PR TITLE
Charts that can be updated in place, generalised data collections

### PR DESCRIPTION
**Pull request by elayabharath**

### CODE REVIEW

File: lib/model.yaml [hunk 1]
```diff
@@ -4,9 +4,8 @@ Tab:
   charts: "array Chart"
 
 Chart:
-  data_path: DataPath
+  data_collection: "string"
```
Comment (marianamarasoiu) [inline lib/model.yaml:7]: I know it's a bit tedious, but the Dart recommended style is to use camel case capitalisation. I think it's preferable for the config to be in underscore though, so let's rename this to camelcase for the generated dart model:
```suggestion
  data_collection: "string dataCollection"
```

File: lib/controller.dart [hunk 2]
```diff
@@ -300,463 +280,162 @@ void loadGeoMapsData() async {
...
+          var collectionToIterate =
+              _dataCollections[currentFilter.data_collection].values;
+          var dates = collectionToIterate
+              .map((e) => e[currentFilter.key])
+              .toList()
+                ..sort();
+          _filtersMap[i][j].options = [dates.first, dates.last];
+          if (_filtersMap[i][j].value == '') {
+            _filtersMap[i][j].value =
+                '${dates.first.split("T")[0]}_${dates.last.split("T")[0]}';
```
Comment (marianamarasoiu) [inline lib/controller.dart:320]: it seems that you need to do this a lot - can we maybe create a `getDate(String datetime)` that implements this logic of extracting the date?

**Review by marianamarasoiu — APPROVED** [review]
LGTM!

File: lib/model.yaml (hunk 1 above)
Comment (elayabharath) [reply lib/model.yaml:7]: Ahh this is cool, didnt know this was an option. Fixed

File: lib/controller.dart (hunk 2 above)
Comment (elayabharath) [reply lib/controller.dart:320]: Yep sure, will refactor to a separate function

**Comment by elayabharath** [comment]
Thanks @marianamarasoiu

**Pull request merged**